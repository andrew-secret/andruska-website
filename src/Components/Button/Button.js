import React from 'react';
import classnames from 'classnames';
import styles from './Button.module.sass';

const Button = props => {
  const buttonClasses = classnames(styles.buttonStyle, props.className);
  return (
    <button className={buttonClasses} type={props.type} onClick={props.onClick}>
      {props.label}
      <i className="material-icons">{props.icon}</i>
    </button>
  );
};

export default Button;

Button.defaultProps = {
  type: 'button',
  label: 'This is a button',
};
