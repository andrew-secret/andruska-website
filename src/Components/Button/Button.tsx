import * as React from 'react';
import * as classnames from 'classnames';

const styles = require('./Button.module.sass');

interface Props {
  className?: string;
  label: string;
  type: 'button' | 'submit';
  icon?: string;
  onClick?: () => void;
}

export const Button: React.SFC<Props> = props => {
  const buttonClasses = classnames(styles.buttonStyle, props.className);
  return (
    <button className={buttonClasses} type={props.type} onClick={props.onClick}>
      {props.label}
      <i className="material-icons">{props.icon}</i>
    </button>
  );
};
