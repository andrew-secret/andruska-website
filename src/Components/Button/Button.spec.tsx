import * as React from 'react';
import {
  renderComponentShallow,
  renderComponentShallowForSnapshot,
} from '../../lib/test-helpers';
import { Button } from './Button';

describe('Button', () => {
  it('Button renders correctly', () => {
    const wrapper = renderComponentShallow(
      <Button label="some button" type="button" />
    );
  });
  it('renders with snapshots', () => {
    const tree = renderComponentShallowForSnapshot(
      <Button label="some button" type="submit" />
    );
    expect(tree).toMatchSnapshot();
  });
});
