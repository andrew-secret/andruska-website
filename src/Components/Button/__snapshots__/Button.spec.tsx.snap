// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders with snapshots 1`] = `
<button
  className=""
  onClick={undefined}
  type="submit"
>
  some button
  <i
    className="material-icons"
  />
</button>
`;
